YieldHandler.sol

```diff
@@ -2,69 +2,61 @@
 pragma solidity 0.8.9;
 
 import "./ITopUpHandler.sol";
+import "@yield-protocol/vault-interfaces/DataTypes.sol";
+import "@yield-protocol/vault-interfaces/ICauldron.sol";
+import "@yield-protocol/vault-interfaces/ILadle.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I128.sol";
+import "@yield-protocol/utils-v2/contracts/token/IERC20.sol";
 
-struct Vault {
-    address owner;
-    bytes6 seriesId;
-    bytes6 ilkId;
-}
-
-interface Ladle {
-    function pour(bytes12 vaultId_, address to, int128 ink, int128 art)
-        external payable;
-}
-
-interface Cauldron {
-    function assets(bytes6) external view returns (address);
-    function vaults(bytes12 vaultId) external view returns (Vault memory);
-    function level(bytes12 vaultId)
-        external
-        returns (int256);
-}
 
 contract YieldHandler is ITopUpHandler {
-    Cauldron public immutable cauldron;
-    Ladle public immutable ladle;
-    Vault public vault;
+    using CastU256I128 for uint256;
+
+    DataTypes.Vault vault;
+    ICauldron cauldron;
+    ILadle ladle;
+    IJoin join;
+    IERC20 collateral;
 
-    constructor(address cauldronAddress, address ladleAddress, bytes12 vaultId) {
-        cauldron = Cauldron(cauldronAddress);
-        ladle = Ladle(ladleAddress);
-        vault = cauldron.vaults(bytes12(vaultId));
+    constructor(ICauldron cauldronAddress, ILadle ladleAddress) {
+        cauldron = cauldronAddress;
+        ladle = ladleAddress;
     }
 
     /**
      * @notice Tops up the account for the protocol associated with this handler
      * This is designed to be called using delegatecall and should therefore
      * not assume that storage will be available
      *
-     * @param account account to be topped up
-     * @param underlying underlying currency to be used for top up
+     * @param account account to be topped up (vaultId)
+     * @param underlying underlying currency to be used for top up (collateral)
      * @param amount amount to be topped up
      * @return true if the top up succeeded and false otherwise
      */
 
     function topUp(
-        address account,
+        bytes32 account,
         address underlying,
         uint256 amount,
-        bool repayDebt
+        bytes memory extra
     ) external returns (bool) {
-        require(account == vault.owner, "Wrong vault!");
-        require(underlying == cauldron.assets(bytes6(vault.seriesId)), "Wrong underlying!");
-        ladle.pour(vault.seriesId, address(0), int128(int256(amount)), 0);
+        vault = cauldron.vaults(bytes12(account));
+        bytes6 baseId = cauldron.series(vault.seriesId).baseId;
```

Comment:
The `base` is what is borrowed, the `ilk` is the collateral posted as a guarantee for the borrowing. You can actually remove this line.

Reply:
Ok. There was a bit of confusion perhaps with this comment then. 
![image](https://user-images.githubusercontent.com/24715302/163922210-9b208da8-6240-43cb-96b4-fc3a1858313f.png)